### PR TITLE
[SAIC-603] NoneType snet/ext_snet handling

### PR DESCRIPTION
**Pull request by raiesmh08**
From "https://github.com/MirantisWorkloadMobility/CloudFerry/blob/master/
cloudferrylib/os/network/neutron.py#L1201"

we can notice that, during hit of this portion of code, if src_net and ex_snet
is None then it will fail always.

This patch puts fix for it.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Merge check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1408/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:1395: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1405: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1407: [W0311(bad-indentation), ] Bad indentation. Found 8 spaces, expected 4
cloudferrylib/os/network/neutron.py:1408: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1409: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1410: [W0311(bad-indentation), ] Bad indentation. Found 12 spaces, expected 8
cloudferrylib/os/network/neutron.py:1411: [W0311(bad-indentation), ] Bad indentation. Found 16 spaces, expected 12
cloudferrylib/os/network/neutron.py:1412: [W0311(bad-indentation), ] Bad indentation. Found 20 spaces, expected 16
cloudferrylib/os/network/neutron.py:81: [W1202(logging-format-interpolation), NeutronNetwork.read_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/network/neutron.py:149: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:259: [W0621(redefined-outer-name), NeutronNetwork.get_network] Redefining name 'network' from outer scope (line 21)
cloudferrylib/os/network/neutron.py:727: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method
cloudferrylib/os/network/neutron.py:831: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_vips] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:856: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_members] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:887: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_monitors] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:948: [W1201(logging-not-lazy), NeutronNetwork.upload_lb_pools] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1190: [W1201(logging-not-lazy), NeutronNetwork.upload_routers] Specify string format arguments as logging function parameters
cloudferrylib/os/network/neutron.py:1207: [W0101(unreachable), NeutronNetwork.add_router_interfaces] Unreachable code
cloudferrylib/os/network/neutron.py:1327: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1334: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.
cloudferrylib/os/network/neutron.py:1338: [W1504(unidiomatic-typecheck), NeutronNetwork.get_resource_hash] Using type() instead of isinstance() for a typecheck.

File: cloudferrylib/os/network/neutron.py [hunk 1]
```diff
@@ -1198,16 +1198,17 @@ def add_router_interfaces(self, src_router, dst_router,
...
-            LOG.debug("Adding subnet '%s' to router '%s'", snet_id,
-                      dst_router['name'])
-            self.neutron_client.add_interface_router(
-                dst_router['id'],
-                {"subnet_id": ex_snet['id']})
+                LOG.debug("Adding subnet '%s' to router '%s'", snet_id,
+                          dst_router['name'])
+                self.neutron_client.add_interface_router(
+                    dst_router['id'],
+                    {"subnet_id": ex_snet['id']})
```
Comment (roman-verchikov) [inline cloudferrylib/os/network/neutron.py]: These lines will never be executed because of `continue` right above

Comment (raiesmh08) [inline cloudferrylib/os/network/neutron.py]: Yes, my mistake. I will be update it in next patch tomorrow IST :)

**Pull request closed without merging**